pkg/agent/agent.go

```diff
@@ -16,6 +16,10 @@ import (
 
 func receiveNewSeccompFile(resolver registry.ResolverFunc, sockfd int) (*registry.Registry, *os.File, error) {
 	MaxNameLen := 4096
+
+	// We want to receive an int (a file descriptor).
+	// 4 = sizeof(int) in most architectures
+	// We should probably revisit this when porting
```

Comment:
Suggested change:

```suggestion
	// File descriptors over SCM_RIGHTS are 'int' according to "man cmsg".
	// The unix golang package assumes that a file descriptor is a int32, see:
	// https://github.com/golang/sys/blob/68d13333faf2/unix/sockcmsg_unix.go#L66-L73
	// On Linux and Windows, `int` is always 32 bits, so it's fine:
	// https://en.wikipedia.org/wiki/64-bit_computing#64-bit_data_models
```

Reply:
You could add `// +build linux` at the top of the file, since this will work only on Linux.

Reply:
Not sure I follow, isn't all of this Linux only? Why add that tag? Or add them to all files, you mean?

Reply:
Adding the tag would make sure that people don't try to use this SCM_RIGHTS code on other OSes because I am not sure it would work elsewhere.

seccompagent is Linux only, but sometimes people on Windows/macOS like to be able to run unit tests with "go test" on specific packages (even if they can't compile the full project), so it could be useful to add the Linux tag only on package that actually require Linux.

I see that [runc's cmsg.go](https://github.com/opencontainers/runc/blob/249bca0a1316129dcd5bd38b5d75572274181cb5/libcontainer/utils/cmsg.go#L36) uses `unix.CmsgSpace(4)` and has the Linux tag, so I would just do the same tag.

SCM_RIGHTS can actually work on other Unixes:
https://github.com/golang/sys/blob/68d13333faf2/unix/sockcmsg_unix.go#L5-L6
```
//go:build aix || darwin || dragonfly || freebsd || linux || netbsd || openbsd || solaris || zos
// +build aix darwin dragonfly freebsd linux netbsd openbsd solaris zos
```
But I suggest we don't complicate this and just add a Linux tag.

Reply:
Ohh, haven't thought of people using other OSs :see_no_evil: ! Good idea, will add that!

Reply:
Added locally, will be in the next push